Action Manager: A few automated tests execute before Action Manager has created menus

**Describe the bug**
A few automated tests execute before Action Manager has created menus. The tests fail due to attempting to trigger new level creation via File > New Level which fails as the option isn't present, and no modal widget pops up due to the trigger failure. This affects:

* test_BasicEditorWorkflows_LevelEntityComponentCRUD
* test_BasicEditorWorkflows_GPU_LevelEntityComponentCRUD

```
E   Failed: Test test_BasicEditorWorkflows_LevelEntityComponentCRUD:
E   Test FAILED
E   ------------
E   |  Output  |
E   ------------
E   Starting test BasicEditorWorkflows_LevelEntityComponentCRUD...
E   Test BasicEditorWorkflows_LevelEntityComponentCRUD finished.
E   Report:
E   EXCEPTION raised:
E     Traceback (most recent call last):
E       File "d:\github\o3de\automatedtesting\gem\pythontests\editorpythontesttools\editor_python_test_tools\utils.py", line 328, in start_test
E         test_function()
E       File "D:/GitHub/o3de/AutomatedTesting/Gem/PythonTests/editor/EditorScripts/BasicEditorWorkflows_LevelEntityComponentCRUD.py", line 161, in BasicEditorWorkflows_LevelEntityComponentCRUD
E         run_test()
E       File "D:\GitHub\o3de\Gems\QtForPython\Editor\Scripts\pyside_utils.py", line 257, in wrapper
E         return run_async(result)
E       File "D:\GitHub\o3de\Gems\QtForPython\Editor\Scripts\pyside_utils.py", line 236, in run_async
E         return awaitable.result()
E       File "D:/GitHub/o3de/AutomatedTesting/Gem/PythonTests/editor/EditorScripts/BasicEditorWorkflows_LevelEntityComponentCRUD.py", line 90, in run_test
E         active_modal_widget = await pyside_utils.wait_for_modal_widget()
E       File "D:\GitHub\o3de\Gems\QtForPython\Editor\Scripts\pyside_utils.py", line 927, in wait_for_modal_widget
E         return await wait_for(lambda: QtWidgets.QApplication.activeModalWidget(), timeout=timeout)
E       File "D:\GitHub\o3de\Gems\QtForPython\Editor\Scripts\pyside_utils.py", line 203, in wait_for
E         await wait_for_condition(condition, timeout)
E     pyside_utils.EventLoopTimeoutException
E   Test result:  FAILURE
```

**Steps to reproduce**
1. Update your AutomatedTesting\Registry\editorpreferences.setreg file to include the following under O3DE:
```
		"ActionManager": {
			"EnableNewActionManager": true
		}
```
2. Run the affected test class:
```
python.cmd -m pytest [Path to AutomatedTesting]/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode --build-directory [Path to build directory]
```

**Expected behavior**
Tests run after Action Manager menus have been setup.

**Actual behavior**
Tests run before menus are setup. Watching the tests run, you can see a brief blip of the menus, and the only option present is AWS. Issue seems similar to #12930.

**Found in Branch**
development@ ff1b1717054132f63d55f5b7877061d5cf1e669e

--- a/AutomatedTesting/Gem/PythonTests/editor/EditorScripts/BasicEditorWorkflows_LevelEntityComponentCRUD.py
+++ b/AutomatedTesting/Gem/PythonTests/editor/EditorScripts/BasicEditorWorkflows_LevelEntityComponentCRUD.py
@@ -68,6 +68,7 @@ def BasicEditorWorkflows_LevelEntityComponentCRUD():
         import azlmbr.bus as bus
         import azlmbr.editor as editor
         import azlmbr.entity as entity
+        import azlmbr.legacy.general as general
         import azlmbr.math as math
         import azlmbr.paths as paths
 
@@ -85,6 +86,12 @@ def BasicEditorWorkflows_LevelEntityComponentCRUD():
         # 1) Create a new level
         lvl_name = "tmp_level"
         editor_window = pyside_utils.get_editor_main_window()
+
+        # The action manager doesn't register the menus until the next system tick, so need to wait
+        # until the menu bar has been populated
+        general.idle_enable(True)
+        await pyside_utils.wait_for_condition(lambda: len(editor_window.menuBar().actions()) > 1)
+
         new_level_action = pyside_utils.get_action_for_menu_path(editor_window, "File", "New Level")
         pyside_utils.trigger_action_async(new_level_action)
         active_modal_widget = await pyside_utils.wait_for_modal_widget()

--- a/AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_EditMenuOptions.py
+++ b/AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_EditMenuOptions.py
@@ -26,6 +26,7 @@ def Menus_EditMenuOptions_Work():
     :return: None
     """
 
+    import azlmbr.legacy.general as general
     import editor_python_test_tools.hydra_editor_utils as hydra
     import pyside_utils
     from editor_python_test_tools.utils import Report
@@ -51,7 +52,6 @@ def Menus_EditMenuOptions_Work():
         ("Modify", "Transform Mode", "Scale"),
         ("Editor Settings", "Global Preferences"),
         ("Editor Settings", "Editor Settings Manager"),
-        ("Editor Settings", "Keyboard Customization", "Customize Keyboard"),
         # The following menu options are temporarily disabled due to https://github.com/o3de/o3de/issues/6746
         #("Editor Settings", "Keyboard Customization", "Export Keyboard Settings"),
         #("Editor Settings", "Keyboard Customization", "Import Keyboard Settings"),
@@ -59,6 +59,11 @@ def Menus_EditMenuOptions_Work():
 
     # 1) Open an existing simple level
     hydra.open_base_level()
+
+    # The action manager doesn't register the menus until the next system tick, so need to wait
+    # until the menu bar has been populated
+    general.idle_enable(True)
+    general.idle_wait_frames(1)
 
     # 2) Interact with Edit Menu options
     editor_window = pyside_utils.get_editor_main_window()

--- a/AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_ViewMenuOptions.py
+++ b/AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Menus_ViewMenuOptions.py
@@ -26,20 +26,15 @@ def Menus_ViewMenuOptions_Work():
     :return: None
     """
 
+    import azlmbr.legacy.general as general
     import editor_python_test_tools.hydra_editor_utils as hydra
     import pyside_utils
     from editor_python_test_tools.utils import Report
 
     view_menu_options = [
-        ("Center on Selection",),
-        ("Show Quick Access Bar",),
-        ("Layouts", "Component Entity Layout",),
-        ("Layouts", "Save Layout",),
         ("Viewport", "Go to Position"),
         ("Viewport", "Center on Selection"),
         ("Viewport", "Go to Location"),
-        ("Viewport", "Remember Location"),
-        ("Viewport", "Switch Camera"),
         ("Viewport", "Show Helpers"),
         ("Viewport", "Show Icons"),
         ("Refresh Style",),
@@ -47,6 +42,11 @@ def Menus_ViewMenuOptions_Work():
 
     # 1) Open an existing simple level
     hydra.open_base_level()
+
+    # The action manager doesn't register the menus until the next system tick, so need to wait
+    # until the menu bar has been populated
+    general.idle_enable(True)
+    general.idle_wait_frames(1)
 
     # 2) Interact with View Menu options
     editor_window = pyside_utils.get_editor_main_window()

